src/common/random.h

```diff
@@ -56,5 +56,18 @@ void shuffle_portable(RandomIt begin, RandomIt end, std::mt19937_64 &rng)
   }
 }
 
+/// Returns a random element between the elements in [start, end] will use the random number generator provided as g
+template<typename Iter, typename RandomGenerator>
+Iter select_randomly(Iter start, Iter end, RandomGenerator& g) {
+    std::uniform_int_distribution<> dis(0, std::distance(start, end) - 1);
```

Comment:
I think this needs a special case for `start == end` that doesn't do anything (i.e. just returns `start`), which probably doesn't work and even if it did, will do the wrong thing in `std::advance`.